src/NServiceBus.NHibernate/SharedSession/SharedSessionConfig.cs

```diff
@@ -0,0 +1,21 @@
+namespace NServiceBus.Persistence.NHibernate
+{
+    using Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+
+    /// <summary>
+    /// Deduplication configuration extensions.
+    /// </summary>
+    public static class SharedSessionConfig
+    {
+        /// <summary>
+        /// Disables sharing a connection between SQL Server transport and NHibernate persistence. Intended to be used in conjunction with Outbox only.
+        /// </summary>
+        /// <param name="persistenceConfiguration"></param>
+        public static PersistenceExtentions<NHibernatePersistence> DoNotShareTransportConnection(this PersistenceExtentions<NHibernatePersistence> persistenceConfiguration)
```

Comment:
The main idea behind this setting is to be able to use Outbox for SQL Server transport, In such case we should not share the transport connection. Instead, we're using a `TransactionScope` created by `OutboxDeduplicationBehavior` to guarantee that the changes done by user, in saga storage and outbox storage are atomic. We're relying on the fact that each of these storages opens its own session within that `TransactionScope` but since these sessions are short-lived, they do not overlap with each other (which would cause DTC escalation).

That said, I *think* (not 100% sure) we can use the same approach for non-SQL Server transports + outbox configurations and get rid of `NHibernateStorageContext`.